src/neo/Consensus/ConsensusService.cs

```diff
@@ -185,8 +187,41 @@ private void InitializeConsensus(byte viewNumber)
             {
                 ChangeTimer(TimeSpan.FromMilliseconds(Blockchain.MillisecondsPerBlock << (viewNumber + 1)));
             }
+
+            // Try to speed up consensus if cached future payloads exists
+            if (context.HasFuturePayloads())
+            {
+                TryToConsumeFuturePayload(context.FuturePreparationPayloads);
+                if (context.HasFuturePayloads())
+                    TryToConsumeFuturePayload(context.FutureCommitPayloads);
+                if (context.HasFuturePayloads())
+                    TryToConsumeFuturePayload(context.FutureChangeViewPayloads);
+                if (context.HasFuturePayloads())
+                    TryToConsumeFuturePayload(context.FutureRecoveryPayloads);
+            }
         }
 
+        private void TryToConsumeFuturePayload(ConsensusPayload[] payloadsArray)
+        {
+            for (int p = 0; p < payloadsArray.Length; p++)
```

Comment:
`foreach`?

Reply:
Can be, let's modify.

Reply:
How to set the variable of the `foreach` to null?

`foreach(ConsensusPayload payload in payloadsArray)`

Because current we do `payloadsArray[p] = null;`